assets/db_schema/all_entities.iuml

```diff
@@ -0,0 +1,343 @@
+' Zero or One	|o--
+' Exactly One	||--
+' Zero or Many	}o--
+' One or Many	}|--
+' One to Many	||--o{
+' Many to One	}o--||
+
+@startuml(id=swap)
+
+entity "swap" as swap {
+    id : <<pk>>,
+    ---
+    local_swap_id : <<unique>>,
+    role,
+    counter_party_peer_id
+}
+
+entity "shared_swap_id" as shared_swap_id {
+    id : <<pk>> <<fk>>,
+    ---
+    shared_swap_id
+}
+
+entity "address_hint" as address_hint {
+    id : <<pk>>
+    ---
+    counter_party_peer_id : <<unique>> <<fk>>,
+    address_hint
+    }
+
+entity "secret_hash" as secret_hash {
+    swap_local_id : <<pk>> <<fk>>,
+    ---
+    secret_hash
+}
+
+@enduml
+
+@startuml(id=swap_connections)
+
+shared_swap_id |o--|| swap
+address_hint |o--|{ swap
+secret_hash |o--|| swap
+
+@enduml
+
+@startuml(id=han_bitcoin_swap_protocols)
+
+entity "han_bitcoin_swap_detail" as han_bitcoin {
+    han_bitcoin_swap_detail_id : <<pk>>,
+    ---
+    swap_local_id : <<fk>>,
```

Comment:
Only a minor thing, but can you actually create a foreign_key relationship on something that is not a primary key? I would have expected these to point to `swap.id`.

Reply:
Yes, `swap.id` should be used for referencing inside the database! @bonomat we discussed this yesterday, the database should use the database internal identifier (the PK) for identifying the swap rather than the `swap_local_id`.

Reply:
Yes, this should be `swap.id`.

Reply:
I hope I caught all.